Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/PaywallViewAPI.swift

```diff
@@ -44,49 +44,102 @@ struct App: View {
+                                    fonts: self.fonts,
+                                    purchaseStarted: self.purchaseStarted,
                                     purchaseCompleted: self.purchaseOrRestoreCompleted,
                                     restoreCompleted: self.purchaseOrRestoreCompleted,
                                     onDismiss: self.paywallDismissed)
-            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "", offering: self.offering, fonts: self.fonts,
+            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "",
+                                    fonts: self.fonts,
+                                    purchaseStarted: nil,
+                                    purchaseCompleted: self.purchaseOrRestoreCompleted,
+                                    restoreCompleted: self.purchaseOrRestoreCompleted,
+                                    onDismiss: self.paywallDismissed)
+            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "",
+                                    offering: self.offering,
+                                    fonts: self.fonts,
                                     purchaseCompleted: self.purchaseOrRestoreCompleted,
-                                    purchaseCancelled: self.purchaseCancelled,
                                     restoreCompleted: self.purchaseOrRestoreCompleted,
                                     onDismiss: self.paywallDismissed)
-            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "", offering: self.offering, fonts: self.fonts,
+            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "",
+                                    offering: self.offering,
+                                    fonts: self.fonts,
                                     purchaseStarted: self.purchaseStarted,
                                     purchaseCompleted: self.purchaseOrRestoreCompleted,
                                     purchaseCancelled: self.purchaseCancelled,
                                     restoreCompleted: self.purchaseOrRestoreCompleted,
                                     onDismiss: self.paywallDismissed)
-            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "", offering: self.offering, fonts: self.fonts,
+            .presentPaywallIfNeeded(requiredEntitlementIdentifier: "",
+                                    offering: self.offering,
+                                    fonts: self.fonts,
                                     purchaseStarted: self.purchaseStarted,
                                     purchaseCompleted: self.purchaseOrRestoreCompleted,
                                     purchaseCancelled: self.purchaseCancelled,
                                     restoreStarted: self.restoreStarted,
                                     restoreCompleted: self.purchaseOrRestoreCompleted,
+                                    purchaseFailure: self.failureHandler,
```

Comment:
Maybe add one with all the parameters but nil values? Same for the footer.

Reply:
good idea